codechecker_lib/session_manager.py

```diff
@@ -288,11 +313,12 @@ def getAuthString(self, host, port):
 
     def saveToken(self, host, port, token, destroy=False):
         if destroy:
-            del self.__save["tokens"][host + ":" + port]
+            del self.__tokens[host + ":" + port]
         else:
-            self.__save["tokens"][host + ":" + port] = token
+            self.__tokens[host + ":" + port] = token
 
-        session_cfg_file = os.path.join(os.path.expanduser("~"),
-                                        ".codechecker_passwords.json")
-        with open(session_cfg_file, 'w') as scfg:
-            json.dump(self.__save, scfg, indent=2, sort_keys=True)
+        with open(self.token_file, 'w') as scfg:
+            fcntl.lockf(scfg, fcntl.LOCK_EX)
```

Comment:
Is this way of locking unix specific? Will it work on OSX and maybe later on Windows?